internal/ccsmp/ccsmp_core.go

```diff
@@ -567,10 +570,11 @@ func NewSessionDispatch(id uint64) (*SolClientSessionRxMsgDispatchFuncInfo, uint
 	// CGO defines void* as unsafe.Pointer, however it is just arbitrary data.
 	// We want to store a number at void*
 	ptr := uintptr(id)
+	// this function should be deprecated in favor allocating the dispatch struct on the C heap
```

Comment:
Actually this does eventually allocate on C heap.  If we want to allocate the struct in Go, we don't have to use C.malloc() we can use a Go Struct as long as we don't put any pointers in it ... but then we'd still need a special C function to convert that struct to a dispatch struct.

Let's not comment this should be deprecated now.

Reply:
Okay, noted.